src/opt/test/test_renderer.cc

```diff
@@ -54,17 +57,24 @@ void TestRendererPixelAccuracy(const Camera& camera) {
   // Create a mesh with a defined color and depth for each pixel in the camera.
   std::uniform_real_distribution<> depth_distribution(0.2f, 20.0f);
   std::uniform_int_distribution<> color_distribution(0, 255);
+
+  const int grid_width = camera.width()/kStep + 1;
+  const int grid_height = camera.height()/kStep + 1;
   
   pcl::PointCloud<pcl::PointXYZRGB> mesh_vertex_cloud;
-  mesh_vertex_cloud.reserve(camera.width() * camera.height());
-  for (int y = 0; y < camera.height(); ++ y) {
-    for (int x = 0; x < camera.width(); ++ x) {
+  mesh_vertex_cloud.reserve(grid_width * grid_height);
+  for (int y = 0; y <= camera.height(); y += kStep) {
```

Comment:
Does that actually work with x or y equal to width or height? Doesn't it below potentially use the camera's undistortion lookup, which is only of size width * height (thus this should cause an out-of-bounds access)?

Reply:
The vector is clamped to fit inside the image.

The rational behind this is to have a mesh grid covering the whole image instead of `image size / grid size -1`

The point at (width, height) is wrongly un istorted but in the end we don't check it, we just want the other vertices to be linked with multiple triangles. Otherwise, the point will not be painted because of how opengl paints pixels in [0, x[ interval (x not included)

Reply:
Ah, I missed that UnprojectFromImageCoordinates() internally clamps the coordinates. Thanks for clarifying.